asmcomp/power/emit.mlp

```diff
@@ -84,6 +81,14 @@ let rodata_space =
 let toc_space =
   " .section \".toc\",\"aw\"\n"
 
+(* Output .text section directive, or named .text.caml.<name> if function
+   sections are enabled and supported on the target system. *)
+let emit_named_text_section func_name =
+  if !Clflags.function_sections then
+    `	.section .text.caml.{emit_symbol func_name},{emit_string_literal "ax"},@progbits\n`
+  else
+    `	.section .text\n`
```

Comment:
Can we move this function to `Emitaux` to share it among all backends that use function sections?

Reply:
amd64 has a more complicated logic since it supports non-ELF binary targets. There is also the annoying choice of character before the section content type (`progbits`): theoretically, this character should be `%` on arm and `@` everywhere else, but so far `gas` and `clang` have been permissive enough to allow the use of `%` everywhere.

Reply:
Even if the code sharing is not perfect, I think we should still do what we can (duplicated code among several backends always ends up getting out of sync): we keep the more complicated logic for `amd64` in place, but call the function in `Emitaux` for the ELF case. As for the separating character we could pass it as an argument. What do you think?

Reply:
Sure. I'll do this.